climate_index_collection/data_specs.py

```diff
@@ -0,0 +1,186 @@
+        # check if all signs are either positive or negative or mixed 
+        # and set corresponding values
+    if np.all(diff > 0) :
+        diff_sign = True
+    elif np.all(diff < 0) :
+        diff_sign = False
+    else :
+        diff_sign = None
+
+    return dict(
+            mini = minimum,
+            mini_pos = np.where(data == minimum)[0],
+            maxi = maximum,
+            maxi_pos = np.where(data == maximum)[0],
+            size = np.size(data), 
+            diff_constant = diff_is_constant,
+            diff_sign = diff_sign,
+            diff_change_position = diff_change_position,
+            diff_change_values = diff_change_values,
+            diff = diff,
+            )
+
+def latitude_longitude_specs(dobj, lat_name ='lat', lon_name='lon') :
+    """This function will return information about the latitude and longitude 
+    structure of the data.
+    The information for both are calculated with get_spacial_dimension_specs.
+    
+    Parameters
+    ----------
+    dobj: xarray.DataArray
+        Data for which the latitude and longitude informations shall be given.
+    lat_name: str
+        Name of the latitude dimension. Defaults to "lat".
+    lon_name: str
+        Name of the longitude dimension. Defaults to "lon".
+    Returns
+    -------
+    dict with keys:
+        lat_specs : dict containing specifics for latitude dimension
+        lon_specs : dict containing specifics for longitude dimension
+    """
+
+    # rcalculate the specifications for latitude and longitude and write into dictonary
+    return dict(
+        lat_specs = get_spacial_dimension_specs(dobj[lat_name].values), 
+        lon_specs = get_spacial_dimension_specs(dobj[lon_name].values)
+        )
+
+def sel_latitude_longitude_slice(dobj, LatBounds = None, LonBounds = None, lat_name = 'lat', lon_name = 'lon'):
+    """Returns latitude and longitude slices for given latitude and longitude bounds 
+    dependent on the data sets latitude and longitude structure.
+    The informations given about increasing or decreasing latitude and longitude values 
+    are derived by the latitude_longitude_specs function.
+    
+    Parameters
+    ----------
+    dobj: xarray.DataArray
+        Data for which the latitude and longitude informations shall be given.
+    LatBounds : list, tupel or np.array
+        Latitude Bounds for which the correct slice shall be obtained.
+        Only max and min values will be used ! 
+    LonBounds : list, tupel or np.array
+        Longitude Bounds for which the correct slice shall be obtained.
+        Only max and min values will be used !
+    lat_name: str
+        Name of the latitude dimension. Defaults to "lat".
+    lon_name: str
+        Name of the longitude dimension. Defaults to "lon".
+
+    Returns
+    -------
+    slice :
+        Latitude slice
+    slice : 
+        Longitude slice
+    """
+    
+    LatLondSpecs = latitude_longitude_specs(dobj = dobj)
```

Comment:
Let's use `snake_case` variable naming here. (In general, follow https://peps.python.org/pep-0008/ wherever possible.)